site/bps/BP-42-listledgers-newapi.md

```diff
@@ -0,0 +1,70 @@
+---
+title: "BP-42: New Client API for listing ledgers "
+issue: https://github.com/apache/bookkeeper/<issue-number>
+state: "Under Discussion"
+release: "x.y.z"
+---
+
+### Motivation
+
+The new Client API (`org.apache.bookkeeper.client.api.BookKeeper`) aims to replace obsolete BookKeeperAdmin API (`org.apache.bookkeeper.client.BookKeeperAdmin`) but some features are not implemented yet. 
+For example, it does not expose a method to list available ledgers, comparable to `BookKeeperAdmin#listLedgers()`.
+
+The goal is to extends the Client API for listing ledgers.
+
+Moreover current method  `BookKeeperAdmin#listLedgers()` does not kindly handle the network errors; for instance, if an IOException occurs during iterator flow, the error is not visible to the caller and the iteration is stopped (e.g. hasNext will return false). This can cause vary issues.
```

Comment:
> does not kindly handle

what about "does not report errors from the metadata driver"

> This can cause vary issues.

you can drop this statement